src/test/java/com/spotify/futures/CompletableFuturesTest.java

```diff
@@ -413,6 +415,62 @@ public void joinList_containsNull() throws Exception {
     stream.collect(joinList());
   }
 
+  @Test
+  public void joinMap_empty() throws Exception {
+    final Map<String, String> result = Stream.<String>of()
+            .collect(joinMap(identity(), CompletableFuture::completedFuture))
+            .get();
+
+    assertThat(result, not(nullValue()));
+    assertThat(result.values(), hasSize(0));
+  }
+
+  @Test
+  public void joinMap_one() throws Exception {
+    final Map<String, String> result = Stream.of("a")
+            .collect(joinMap(identity(), k -> completedFuture(k + "v")))
+            .get();
+
+    assertThat(result.values(), hasSize(1));
+    assertThat(result.values(), contains("av"));
+    assertThat(result.keySet(), contains("a"));
+  }
+
+  @Test
+  public void joinMap_two() throws Exception {
+    final CompletableFuture<String> a = completedFuture("hello");
+    final CompletableFuture<String> b = completedFuture("world");
+
+    final Map<String, String> result = Stream.of("hello", "world")
+            .collect(joinMap(e -> "k " + e, e -> completedFuture("v " + e)))
+            .get();
+    assertThat(result.keySet(), contains("k hello", "k world"));
+    assertThat(result.values(), contains("v hello", "v world"));
```

Comment:
Technically, I think it would be more idiomatic to use `hasEntry(k, v)` matchers here, it also prevents the unlikely scenario that the returned map contains `"k hello" -> "v world"` entries and vice versa.  But not a big deal as that's very unlikely.